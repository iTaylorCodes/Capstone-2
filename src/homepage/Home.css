.home-hero {
  display: flex;
  justify-content: center;
  flex-direction: column;
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: -1;
  background-image: url("./Neighborhoodz-bg.jpg");
  background-size: cover;
  background-position: center center;
}

.Home-logo {
  width: 60px;
  height: 60px;
}

.Home-heading {
  text-shadow: -0.5px 0 black, 0 0.5px black, 0.5px 0 black, 0 -0.5px black;
}

.Home-heading h3 {
  font-weight: bold;
  color: #47c547;
}

.Home-heading h1 {
  font-weight: bold;
}
