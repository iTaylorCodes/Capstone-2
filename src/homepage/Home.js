const Home = () => {
  return <div>Temporary Neighborhood Homepage</div>;
};

export default Home;
