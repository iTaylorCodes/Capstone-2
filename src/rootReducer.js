import { v4 as uuidv4 } from "uuid";

const INITIAL_STATE = {
  accountWasDeleted: false,
  recentSearches: [],
};

const rootReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case "ACCOUNT_DELETED":
      return {
        ...state,
        accountWasDeleted: true,
      };

    case "RESET_ACCOUNT_DELETED":
      return {
        ...state,
        accountWasDeleted: false,
      };

    case "NEW_SEARCH":
      if (state.recentSearches.length >= 5) {
        state.recentSearches.shift();
      }

      return {
        ...state,
        recentSearches: [
          ...state.recentSearches,
          {
            city: action.searchTerm,
            image: action.image,
            scores: action.scores,
            id: uuidv4(),
          },
        ],
      };

    default:
      return state;
  }
};

export default rootReducer;
