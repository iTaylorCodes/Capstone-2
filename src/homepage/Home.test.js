import { render, act, fireEvent } from "@testing-library/react";
import Home from "./Home";
import * as redux from "react-redux";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../rootReducer";

const store = createStore(rootReducer);

const reduxSpy = jest.spyOn(redux, "useSelector");

beforeEach(() => {
  reduxSpy.mockReturnValue({
    accountWasDeleted: false,
    recentSearches: [],
  });
});

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <Home />
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Home />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows alerts for accountWasDeleted", () => {
  act(() => {
    reduxSpy.mockReturnValue({
      accountWasDeleted: true,
      recentSearches: [],
    });

    const { getByText } = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );

    const alert = getByText("Your account has been deleted.");

    expect(alert).toBeTruthy();
  });
});
