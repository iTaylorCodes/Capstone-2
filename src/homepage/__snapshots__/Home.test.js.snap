// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center m-3"
  >
    <div
      class="fw-bold"
      id="alert-container"
    >
      <div
        class="alert alert-warning m-3"
        role="alert"
      >
        <p
          class="mb-0 small"
        >
          Your account has been deleted.
        </p>
      </div>
    </div>
    <div
      class="HomeContent m-3"
    >
      <div>
        <h1
          class="text-primary mb-3"
        >
          Neighborhood
        </h1>
      </div>
      <div
        class="SearchBar d-flex justify-content-center"
      >
        <div
          class="mb-3"
        >
          <form
            class="input-group"
          >
            <div
              class="form-floating"
            >
              <input
                class="form-control ps-5 pe-5"
                id="city-search"
                name="searchTerm"
                placeholder="Search City, State:"
                value=""
              />
              <label
                for="city-search"
              >
                Search City, State:
              </label>
            </div>
            <button
              class="btn btn-primary"
              data-testid="search-button"
              type="submit"
            >
              <i
                class="fa fa-search"
              />
            </button>
          </form>
        </div>
      </div>
      <div
        class="SearchesCarousel container d-flex justify-content-center"
      >
        <div
          class="col-12 col-md-8 col-lg-6 col-xl-5 col-xxl-4 bg-light text-center p-3"
        >
          <h3>
            Recent Searches
          </h3>
          <p>
            Your most recent searches will be shown here.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
